### PR TITLE
fix(highlight): pum matched text respect icase

### LINKED ISSUE #15021: highlight groups PmenuMatch and PmenuMatchSel some buggy
### Steps to reproduce

1. did not expect icase of match:
image

2. did not reflect the change of base:
// os.if -> the `if` highlighted -> changed to os.i which backspace the `f` -> the `i` highlight missed.
image

### Expected behaviour

work.

### Version of Vim

v9.1.493

### Environment

linux

### Logs and stack traces

#14694

### PR DESCRIPTION
**Pull request by glepnir**
Problem: 
1. STRNCMP does not respect the upper case when icase is set of complete item.
2. ins_compl_stop will free all leader even bs just delete the one leader. 

Solution: 
1.use MB_STRNICMP instead 
2. when compl_leader is null use `compl_orig_text` for match.

Fix #15021

### CODE REVIEW

File: src/popupmenu.c [hunk 1]
```diff
@@ -485,7 +485,7 @@ pum_screen_put_with_attr(int row, int col, char_u *text, int textlen, hlf_T hlf)
 	    }
 	}
 	else if (!in_fuzzy && (ptr - text < leader_len)
-			     && (STRNCMP(text, match_leader, leader_len) == 0))
+			     && (MB_STRNICMP(text, match_leader, leader_len) == 0))
```
Comment (zeertzjq) [inline src/popupmenu.c]: This function is called repeatedly when it only needs to be called once. Maybe you can rebase this onto #15020.

File: src/insexpand.c [hunk 2]
```diff
@@ -1448,7 +1448,7 @@ ins_compl_show_pum(void)
     char_u *
 ins_compl_leader(void)
 {
-    return compl_leader;
+    return compl_leader != NULL ? compl_leader : compl_orig_text;
```
Comment (zeertzjq) [inline src/insexpand.c:1451]: Is this change tested?

Comment (glepnir) [reply src/insexpand.c:1451]: missed. should add a bs step.

File: src/popupmenu.c (hunk 1 above)
Comment (glepnir) [reply src/popupmenu.c]: oh nice. i will wait that.

Comment (glepnir) [reply src/popupmenu.c]: mark as draft then.

Comment (chrisbra) [reply src/popupmenu.c]: I have merged it.

**Comment by zeertzjq** [comment]
I force-pushed to your branch to rebase this PR. It also seems that the two screen dumps add in this PR are the same, so I combined them into one.

**Comment by glepnir** [comment]
it's fine :)

**Pull request closed without merging**